server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -553,6 +559,10 @@ public synchronized IndexShard createShard(
                 () -> globalCheckpointSyncer.accept(shardId),
                 retentionLeaseSyncer,
                 circuitBreakerService,
+                this.indexSettings.isSegRepEnabled() && this.indexSettings.isRemoteStoreEnabled()
+                    ? new RemoteBlobStoreInternalTranslogFactory(
+                        repositoriesServiceSupplier.get().repository(clusterService.state().metadata().clusterUUID()))
+                    : new InternalTranslogFactory(),
```

Comment:
Can we skip the resolution of ``repositoriesServiceSupplier`` here unless needed by the IndexService, and instead let the ``RemoteBlobStore`` resolve it when necessary?